numpy/core/tests/test_deprecations.py

```diff
@@ -560,6 +560,24 @@ def test_flatten_array_invalid_string_arg(self):
         self.assert_deprecated(x.flatten, args=("FACK",))
 
 
+class TestArrayDataAttributeAssignmentDeprecation(_DeprecationTestCase):
+    """Assigning the 'data' attribute of an ndarray is unsafe and has in fact
+    been deprecated at the C level since 1.7. Eventually, such assignment
```

Comment:
Hmmm...I was following along with what @njsmith was saying in this issue, since I he figured he knew something more about this than I would.  However, I didn't actually see any error / deprecation though in the C code.